Add project schemas

paul121 opened the pull request:
Closes #37 

Google sheet: https://docs.google.com/spreadsheets/d/1emZBMyvJz5iD_fYJDV7abmkBnbrsbDDIT3OpGdjMSEc/edit?usp=sharing

S4mmyb commented:
* For Terrasos - `administrativeArea`, `socialCulturalValue`, `environmentalAuthority`, `managementAreas` we want enums in the terrasos schema
* For `biomeType`, `bioregion`, `conservationStatus`, `marketType`, `region`, `subWatershed`, and watershed, in which others would want to use these we should define common enums in the same way we did for the taxonomy
* We should change the names to use semantic naming, so `hasManagementArea` as opposed to `managementArea`. 
* For `conservationStatus` we should rename to `iucnConservationStatus` since this is a known standard
* We discussed creating a list of 3rd party project pages with the `nameURL` data type. Then the url would always point to the regen app. This would replace `cfcProjectPage`, `vscProjectPage` and and the way MBS is using `url` currently, and allow for other marketplace pages to be indexed in a lightweight way in the metadata. 
* co-benefits discussion is in issue #34 
* Start and end date can be common, but is required for credit bearing
* For stakeholders, we want to make a class for organizations and one for individuals, or just do one with an enum that you can choose which type you are (so maybe something like "entity"). Then in the project common metadata schema we can just have the roles of range org class

paul121 commented:
Hey @S4mmyb I completed the above checked items and also added addition `**ProjectInfo` schemas for credit classes that have additional class-specific attributes (C01, C02, C03 and cleaned up existing BT01). I'm not including MBS01 in here for now.

Re: the remaining enums mentioned above:
- For Terrasos specific enums I just need to curate a list of all allowed values! WIP
- For other enums we want to make common... With the exception of `marketType`, I worry these values are fairly specific to Terrasos itself? I don't imagine many other projects would use the same enum values for these regions (at least not with the same meaning) that I see used in the [Terrasos app filters](https://terrasos.app.regen.network/projects/1): `pacific`, `orinoco`, `caribbean`, `andean`
  - Is the main issue that these attribute names themselves `bioregion`, `region`, `watershed`, etc will likely want to be used by other projects, but have different & overlapping allowed values?

---

A couple things that came to mind while working on this:

> We discussed creating a list of 3rd party project pages with the nameURL data type. 

So I added a new `hasLinks` attribute with a range of a new [`Link`](https://fbcac258.regen-data-standards.pages.dev/schema/Link) class with `name` and `url` attributes. `C02` projects also had an `offsetProtocol` attribute link that specified a `version`, so I added a [`VersionedLink`](https://fbcac258.regen-data-standards.pages.dev/schema/VersionedLink) class... But I can't help but notice that we already have many other classes that all have `name`, `url`, `version` and `identifier` attributes... ([Methodology](https://fbcac258.regen-data-standards.pages.dev/schema/Methodology), [Credit Protocol](https://fbcac258.regen-data-standards.pages.dev/schema/CreditProtocol) and [Crediting Program](https://fbcac258.regen-data-standards.pages.dev/schema/CreditingProgram))

I wonder if "semantically" these are not actually Methodologies, Credit Protocols or Crediting Programs, but really just "Resource"/"Resource Link"/"Versioned Resource Link" data types that are used as the value of attributes that happen to be named `hasMethodologies`, `hasCreditProtocols`, etc. Does that make sense? Maybe in the future if we do have a true semantic representation of a methodology then the range of `hasMethodologies` could be updated to include either `ResourceLink | Methodology` but I feel like it might be too early to claim these `Methodolgy` and etc class names :thinking: cc @clevinson 

Last, I notice a pattern in that many of the credit classes that need a specific project schema are only adding attributes to track custom "ids" (`cfcProjectId`, `vcsProjectId`, `toucanProjectTokenId`) and other methodology/protocol links (`vcsMethodology` and `offsetProtocol`). These feel like common things many credit class projects may want to provide. I wonder if we could generalize these in some way and add it to the common project info, like a compound ID field with `externalId` and `identifierType` ??

clevinson commented:
Everything in your last comment sounds good to me @paul121. I'll be back in LA next week so happy to jump on a call to discuss any of this if you could use some unblocking to get this PR to a ready-for-review state.

paul121 commented on schema/src/BT01ProjectInfo.yaml:
```diff
@@ -57,69 +78,67 @@ classes:
+        description: Large Tolima Subandean Orobiome
+      TROPICAL_HUMID_ZONOBIOME_TOLIMA_GRANDE:
+        description: Tropical Humid Zonobiome Tolima Grande
+  BT01WatershedTypes:
+    description: The watersheds for BT01 projects.
+    permissible_values:
+      FRIO_RIVER:
+        description: Frío river
+      SUMAPAZ_RIVER_LOWER_WATERSHED:
+        description: Sumapaz river - lower watershed
+  BT01SubWatershedTypes:
+    description: The watersheds for BT01 projects.
+    permissible_values:
+      RIO_FRIO_Y_OTROS_DIRECTOS_AL_CAUCA:
+        description: Río Frío y otros Directos al Cauca
+      RIO_SUMAPAZ:
+        description: Río Sumapaz
```
There are some inconsistencies in watershed/subwatershed enum values. I also wonder if we could just use a single enum for both of these. Ideally we could do some kind of hierarchy but prob not necessary for now.

- Usage of `rio` vs `river`
- I think `Sumpaz river - lowerwater shed` is the _sub-watershed_, and `Rio Sumapaz` is the watershed? (these are the values I saw for [BT01-002 metadata](https://regenscan.netlify.app/project/BT01-002))

S4mmyb commented on schema/src/ProjectInfo.yaml:
```diff
@@ -7,41 +7,136 @@ prefixes:
     slots:
       - name
+      - description
       - url
```
does it make sense to use link here?

paul121 replied:
Interesting. Currently in the UI you must be using the entity name as the text for the link? Allowing there to be different link text might change things a little.

S4mmyb commented on schema/src/ProjectInfo.yaml:
```diff
@@ -55,16 +150,31 @@ classes:
         range: string
         description: The unit of measurement for the quantity.
-  Organization:
+  Link:
```
Link and Versioned Link can be added to the core file, as they will be used broadly. Also if we do use this we should make sure @clevinson's PR (#35 ) is updated accordingly.

S4mmyb commented on schema/src/ProjectInfo.yaml:
```diff
@@ -104,59 +214,22 @@ classes:
-
 enums:
-  ProjectRoleTypes:
+  EntityTypes:
```
I think we should put Entity and Entity Types in it's own file, like we do for Impact or SDG, as Entities, as it will be used in other places as well.

S4mmyb reviewed (commented):
> > For Terrasos - administrativeArea, socialCulturalValue, environmentalAuthority, managementAreas we want enums in the terrasos schema
> 
> I don't think we can do enums for a couple of these because they include multiple values. `environmentalAuthority` currently uses an `Entity` class which allows name, description, url and entity type. And `managementArea` includes a hectare value + project activity enum value.
That makes sense for environmental authority. For `managementArea` they have two categories, preservation and restoration, which have defined values of 0.16 & 0.20 respectively. We're missing preservation in our taxonomy which we can add, and environment restoration can just be mapped to restoration. Is the issue here that we can't map to a `permissible_value` which is multivalued? 

> For the rest of the Terrasos I made enums based on the 2 projects that have anchored metadata on-chain and the `region` and `ecosystem` filter values displayed in the Terrasos app.
Great, that's perfect. 

> Let's hold off on Impact and etc taxonomies because this will require changes in Credit Classes too. This way we can get things merged and have less conflict.
What do you mean here? 

-- 

@paul121 great job! For the most part I think this looks good, but left some comments on structure or specific elements. I think we need to make sure changes to types, like Link, are reflected in PR #35, and vice versa

S4mmyb commented on schema/src/C02ProjectInfo.yaml:
```diff
@@ -0,0 +1,26 @@
+    title: C02 Project Info
+    description: Project info for City Forest Carbon Credit projects.
+    attributes:
+      offsetProtocol:
```
We could also put in restrictions on the referenceID type to specify the protocol

paul121 replied:
Chatting with Sam, we can leave this additional restriction for later.
